authapi/views.py

```diff
@@ -158,16 +158,52 @@ class TeamPermissionViewSet(
     '''Nested viewset to add and remove permissions from teams.'''
     queryset = SeedPermission.objects.all()
     serializer_class = PermissionSerializer
+    permission_classes = (permissions.TeamPermissionPermission,)
 
-    def create(self, request, parent_lookup_seedteam=None, **kwargs):
+    def check_team_permissions(self, request, teamid, orgid=None):
+        old_method = request.method
+        request.method = 'GET'
```

Comment:
The reason this is in the view instead of `TeamPermissionPermission` is that here we have access to `parent_lookup_seedteam` and `parent_lookup_seedteam__organization`, which comes from the URL. We don't have access to those variables in the permission, we would have to manually extract it from the URL, so unfortunately it has to live here.

It would be nice to have a URL nesting package that checks the permissions of each subresource as you go through them, but unfortunately it's not that easy, because as in this case, you're trying to create (`POST`) a new permission, but you're looking at the permissions as if you're getting (`GET`) a team or organization, and there's no way of knowing what type of permissions should apply to each parent resource.

I'll see if I can create a copy of the request, or maybe just a new request, I think we only really check the `user` and `method` in our permission checks.